### ISSUE TITLE
Saving weights of encode layer to be used in classification

### ISSUE THREAD

**Issue by Tinarights** [keras-team/keras#8576]
Hello, Following this code- https://www.snip2code.com/Snippet/913210/Stacked-Denoising-Autoencoder-using-MNIS   I am not sure if it is right or wrong. I have done the following:

`
# Layer-wise pertaining

encoders = []
weights=[]
models=[]
nb_hidden_layers = [784, 600, 500,400]
X_train_tmp = np.copy(X_train)
for i, (n_in, n_out) in enumerate(zip(nb_hidden_layers[:-1], nb_hidden_layers[1:]), start=1):
    print('Training the layer {}: Input {} -> Output {}'.format(i, n_in, n_out))
    # Create AE and training
    ae = Sequential()
    inputs = Input(shape=(n_in,))
   
    encoder = Dense(n_out, activation='relu')(inputs)
    decoder = Dense(n_in,activation='relu')(encoder)
    ae=Model(inputs, decoder)
    ae.compile(loss='mean_squared_error', optimizer='rmsprop')
    ae.fit(X_train_tmp, X_train_tmp, batch_size=batch_size, nb_epoch=nb_epoch)
    
    # Store trainined weight and update training data
    encoder_func = K.function([ae.layers[0].input], [ae.layers[1].output])
    X_train_tmp  = encoder_func([X_train_tmp])[0]
    
    encoders.append(encoder_func)
    m=ae.save('%s_Model.hdf5'%i)
    models.append(m)
    w=ae.save_weights('%s_w.h5'%i)
    weights.append(w)
    

# Fine-turning
`
model = Sequential()

model.add(Dense(nb_hidden_layers[-1], nb_classes, activation='softmax'))

model.compile(loss='categorical_crossentropy', optimizer='rmsprop')
score = model.evaluate(X_test, Y_test, show_accuracy=True, verbose=0)
print('Test score before fine turning:', score[0])
print('Test accuracy after fine turning:', score[1])
model.fit(X_train, Y_train, batch_size=batch_size, nb_epoch=nb_epoch,
          show_accuracy=True, validation_data=(X_test, Y_test))
score = model.evaluate(X_test, Y_test, show_accuracy=True, verbose=0)
print('Test score after fine turning:', score[0])
print('Test accuracy after fine turning:', score[1])
`

Inside the for loop, I train the auto encoding part, after that I want to use it in classification, Is my saving weights correct?
The error came in # Fine-turning where I need to specify input shape to the dense(classification) layer.
I want exactly the features of the last encode layer, how to extract it well?
